superx/information_extractor.py

```diff
@@ -9,123 +9,155 @@
+            elif self.super_name == 'victory':
+                zip_link = 'http://matrixcatalog.co.il/' + zip_link
+
+            request = requests.get(zip_link)
+            content = request.content
+            xml_file = gzip.decompress(content).decode('utf-8')
+            # parses the xml document into a tree
+            tree = ET.fromstring(xml_file)
+            # gets the branch/store ID of current xml file
+            branch_id = self.get_branch_id(tree.getchildren())
+            # gets relevant child
+            items = tree.getchildren()[-1]
+            self.extract_information(items, branch_id)
+            # break  # only use 1 link:
+
+    def extract_information(self, items, current_branch_id):
         item = 'Item'
-
-    for item in items.findall(item):
-        code = int(item.find('ItemCode').text)
-        name = item.find('ItemName').text
-        price = item.find('ItemPrice').text
-        try:
-            item_id = item.find('ItemId').text
-        except AttributeError:
-            item_id = 0
-
-        item_info_formula = """INSERT INTO item_info (itemCode,itemId,itemName) VALUES(%s,%s,%s) ON DUPLICATE KEY UPDATE itemName = itemName """
-        price_table_shufersal = """INSERT INTO price_table (itemCode, shufersal) VALUES(%s, %s)  ON DUPLICATE KEY UPDATE shufersal = %s """
-        price_table_mega = """INSERT INTO price_table (itemCode, mega) VALUES(%s, %s)  ON DUPLICATE KEY UPDATE mega = %s """
-
-        # places the info into the the item_info table
-        cursor.execute(item_info_formula, (code, item_id, name))
-        conn.commit()
-
-        # places the info into the price_table, according to the supermarket
-        if super_name == 'mega':
-            cursor.execute(price_table_mega, (code, price, price))
-        elif super_name == 'shufersal':
-            cursor.execute(price_table_shufersal, (code, price, price))
-        conn.commit()
-
+        bIsWeighted = 'bIsWeighted'
+
+        # different supermarkets use different titles
+        if self.super_name == 'victory':
+            item = 'Product'
+            bIsWeighted = 'BisWeighted'
+
+        for item in items.findall(item):
+            code = int(item.find('ItemCode').text)
+            name = item.find('ItemName').text
+            quantity = item.find('Quantity').text
+            price = item.find('ItemPrice').text
+            is_weighted = False
+            if item.find(bIsWeighted).text == 1:
+                is_weighted = True
+
+            unit_of_measure = 'none'
+            if is_weighted:
+                unit_of_measure = self.convert_unit_name(item.find('UnitQty').text)
+
+            try:
+                item_id = item.find('ItemId').text
+            except AttributeError:
+                item_id = 0
+
+            item_info_formula = """INSERT INTO item_info (itemCode,itemId,itemName) VALUES(%s,%s,%s) ON DUPLICATE KEY UPDATE itemName = itemName """
```

Comment:
this is no longer relevant as we moved to sql alchemy, i have yet to update the queries